nfp/preprocessing/xtb_preprocessor.py

```diff
@@ -11,73 +11,77 @@
     tf = None
 
 from nfp.preprocessing import features
-from nfp.preprocessing.preprocessor import Preprocessor
+from nfp.preprocessing.mol_preprocessor import MolPreprocessor
 from nfp.preprocessing.tokenizer import Tokenizer
 
-class xTBPreprocessor(Preprocessor):
-    def __init__(
-            self,
-            atom_features: Optional[Callable[[rdkit.Chem.Atom], Hashable]] = None,
-            bond_features: Optional[Callable[[rdkit.Chem.Bond], Hashable]] = None,
-            **kwargs,
-    ) -> None:
-        super(xTBPreprocessor, self).__init__(**kwargs)
 
-        #first set of RDKit features for keeping count of type of atom/bond features.
-        if atom_features is None:
-            self.atom_features_xtb = features.atom_features_xtb
-        if bond_features is None:
-            self.bond_features_xtb = features.bond_features_xtb
+class xTBSmilesPreprocessor(MolPreprocessor):
```

Comment:
Because input for the create_nx_graph is smiles, I kept it as xTBSmilesPreprocessor.

Reply:
Oh -- I'd change that then in case you later wanted to use a Mol object (and to reduce the amount of duplicated code). I'll comment below on `create_nx_graph`